src/wallet/scriptpubkeyman.cpp

```diff
@@ -296,7 +296,22 @@ std::unique_ptr<SigningProvider> LegacyScriptPubKeyMan::GetSigningProvider(const
 
 bool LegacyScriptPubKeyMan::CanProvide(const CScript& script, SignatureData& sigdata)
 {
-    return false;
+    if (IsMine(script) != ISMINE_NO) {
+        return true;
+    } else if (HaveCScript(CScriptID(script))) {
+        return true;
+    } else {
+        ProduceSignature(*this, DUMMY_SIGNATURE_CREATOR, script, sigdata);
+        if (!sigdata.signatures.empty()) {
+            // If we could make signatures, make sure we have the private key
```

Comment:
why?

Reply:
It could be watch only (as DUMMY_SIGNATURE_CREATOR works for watch only things), which then causes issues elsewhere when a private key is assumed to exist but doesn't for watch only things. So this prevents those code paths being taken for watch only things.

Reply:
I'm not writing an essay. It is completely unnecessary to describe what DUMMY_SIGNATURE_CREATOR does here as you can just look at the `sign.cpp/h` where it is defined and read the comments and code there yourself. It's less than 10 lines.